### ISSUE TITLE
CA1823 Avoid Unused Fields fires for const fields only used on attributes in local functions

### ISSUE THREAD

**Issue by jkoritzinsky** [dotnet/roslyn-analyzers#7666]
### Analyzer

**Diagnostic ID**: [CA1823](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1823): `Avoid unused private fields`

### Analyzer source

**SDK**: .NET SDK

**Version**: 10.0.100-preview.3.25201.16

### Describe the bug

If a `private const string` field is only used in an attribute inside of a local function, the analyzer reports it as unused.

### Steps To Reproduce

```csharp
class AttrAttribute (string str) : System.Attribute;

class C
{
   private const string Str = "ABC";

   public void Method()
   {
        LocalMethod();
        [Attr(Str)]
        void LocalMethod(){}
   }
}
```

### Expected behavior

No warnings

### Actual behavior

Warning on `C.Str` that it is unused.
